### PR TITLE
AP_Arming cleanup; use common functions

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_Arming/AP_Arming.cpp [hunk 1]
```diff
@@ -367,7 +367,7 @@ bool AP_Arming::gps_checks(bool report)
     }
 
     // check GPSs are within 50m of each other and that blending is healthy
-    if ((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_GPS_CONFIG)) {
+    if ((checks_to_perform & ARMING_CHECK_ALL) || (checks_to_perform & ARMING_CHECK_GPS)) {
```
Comment (WickedShell) [inline libraries/AP_Arming/AP_Arming.cpp]: I was wondering on this one, I noted that it was config the other day, which I thought was a bit weird, but I also thought there might have been a reason it was not put in as part of the normal GPS checks?

File: ArduCopter/AP_Arming.cpp [hunk 2]
```diff
@@ -96,15 +96,10 @@ bool AP_Arming_Copter::rc_calibration_checks(bool display_failure)
 
 bool AP_Arming_Copter::barometer_checks(bool display_failure)
 {
+    bool ret = AP_Arming::barometer_checks(display_failure);
```
Comment (OXINARF) [inline ArduCopter/AP_Arming.cpp]: Any reason to not return false immediately? I think that's how it is done in the other places.

Comment (peterbarker) [inline ArduCopter/AP_Arming.cpp]: Changed.

**Comment by peterbarker** [comment]
@rmackay9 Should checking of the gps blending and consistency be based off `ARMING_CHECK_GPS` or `ARMING_CPS_CONFIG`?

**Comment by rmackay9** [comment]
Hi there, I think that was probably a mistake on my part ... i think it should be part of ARMING_CHECK_GPS.

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**